File: Behavioral/Memento/Java/History.java
// History.java

import java.util.Stack;

public class History {
  private Stack<EditorMemento> history = new Stack<>();

  public void save(EditorMemento memento) {
    if (memento != null) {
      history.push(memento);
    }
  }

  public EditorMemento undo() {
    if (history.size() > 1) {
      history.pop(); // Remove the latest state
    }
    return history.isEmpty() ? null : history.peek(); // Return the previous state
  }
}

File: Behavioral/Memento/CSharp/History.cs
// History.cs

using System.Collections.Generic;

public class History {
  private Stack<EditorMemento> history = new Stack<EditorMemento>();

  public void Save(EditorMemento memento) {
    if (memento != null) {
      history.Push(memento);
    }
  }

  public EditorMemento? Undo() {
    if (history.Count > 1) {
      history.Pop(); // Remove the latest state
    }
    return history.Count > 0 ? history.Peek() : null; // Return the previous state
  }
}
